client/components/domains/use-my-domain/index.jsx

```diff
@@ -0,0 +1,175 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import React, { useCallback, useEffect, useState, useRef } from 'react';
+import { __, sprintf } from '@wordpress/i18n';
+import { connect } from 'react-redux';
+import { Card, Button } from '@automattic/components';
+import { BackButton } from '@automattic/onboarding';
+import { createElement, createInterpolateElement } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+import Gridicon from 'calypso/components/gridicon';
+import FormattedHeader from 'calypso/components/formatted-header';
+import FormFieldset from 'calypso/components/forms/form-fieldset';
+import FormInputValidation from 'calypso/components/forms/form-input-validation';
+import FormButton from 'calypso/components/forms/form-button';
+import FormTextInput from 'calypso/components/forms/form-text-input';
+import page from 'page';
+
+/**
+ * Style dependencies
+ */
+
+import './style.scss';
+/**
+ * Image dependencies
+ */
+import domainIllustration from 'calypso/assets/images/illustrations/domain.svg';
+import { domainUseYourDomain } from 'calypso/my-sites/domains/paths';
+
+function UseMyDomain( { goBack, initialQuery, selectedSite } ) {
+	const [ domainName, setDomainName ] = useState( initialQuery ?? '' );
+	const [ domainNameValidationError, setDomainNameValidationError ] = useState();
+	const domainNameInput = useRef( null );
+	const initialValidation = useRef( null );
+
+	const baseClassName = 'use-my-domain';
```

Comment:
Is this to facilitate reuse of this page in the future or to reduce repetition in the component? I prefer to write CSS classnames in full instead of using variables (i.e. `'use-my-domain__go-back'` vs `baseClassName + '__go-back'`) because I think it's easier to find the components classes are applied to later, but I'm not sure which approach is better 🙂

Reply:
I like doing it this way because it helps to prevent problems because of typos. You can still search on the part of the classname before the `__` to find components. If you think it's a big issue, I can replace the variables with literals.

Reply:
Gotcha, not an issue at all 🙂